src/corehost/cli/setup.cmake

```diff
@@ -60,6 +60,8 @@ endif()
 # containing the reference instead of using definitions from other modules.
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Xlinker -Bsymbolic -Bsymbolic-functions")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--build-id=sha1")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--build-id=sha1")
     add_compile_options(-fstack-protector-strong)
```

Comment:
`file` will print the build id.  It's is also in a note section with the name .note.gnu.build-id.  `readelf -n` can pull it up.

```

matell@matell-ub1604:~/git/core-setup/artifacts/ubuntu.16.04-x64/corehost$ file dotnet
dotnet: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=21196e6acf7429e72b193aac8a9cba8b98761ab1, not stripped
matell@matell-ub1604:~/git/core-setup/artifacts/ubuntu.16.04-x64/corehost$ readelf -n dotnet

Displaying notes found at file offset 0x0000021c with length 0x00000020:
  Owner                 Data size       Description
  GNU                  0x00000010       NT_GNU_ABI_TAG (ABI version tag)
    OS: Linux, ABI: 2.6.32

Displaying notes found at file offset 0x0000023c with length 0x00000024:
  Owner                 Data size       Description
  GNU                  0x00000014       NT_GNU_BUILD_ID (unique build ID bitstring)
    Build ID: 21196e6acf7429e72b193aac8a9cba8b98761ab1
```

We've found this is more likely to be present in coredumps and it also differs per distro, which will be helpful for some other stuff we want to build.